core/src/test/java/com/netflix/iceberg/util/TestBinPacking.java

```diff
@@ -162,6 +168,15 @@ public void testBinPackingLookBack() {
     // 6. [5, 1]
     Assert.assertEquals("1 bin look-back: should merge ones with fives",
         l(l(5, 1), l(5, 1), l(5, 1)), pack(l(5, 1, 5, 1, 5, 1), 8, 1));
+
+    Assert.assertEquals("1 bin look-back: should merge until targetWeight and order them by weight descending",
```

Comment:
One callout is that bins are sorted by weight (descending) only. It does not look at the number of items within the bins itself. I wasn't sure if we wanted to cover this here?

Reply:
This is okay. We have discussed using a minimum weight to handle over-packing.